Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -40,54 +40,78 @@ public struct AbstractInterpreter {
         propertyTypes.removeAll()
         methodSignatures.removeAll()
     }
-    
+
+    /// Get variables with new type which have different type in old state
+    /// Used after block end when some state on stack is removed
+    private func getStateChanges(oldState: VariableMap<Type>, newState: VariableMap<Type>) -> [(Variable, Type)] {
```

Comment:
Since the interpreter now has to answer the `type(of)` query less often (the function is private now), I think it would be quite efficient to store the `stack` as diffs instead of having all variables be present in every stack element. This will mostly (significantly) speed up the `merge` operation, but should also help here, since the changes are immediately obvious if the state is just a diff. Only the `type(of)` function will now have to traverse the stack to find the current type. WDYT?

Reply:
This can also happen as a separate commit though, and I can also do it

Reply:
yeah as we talked before I thought it will be done separately as this is large PR itself